app/src/main/java/tech/relaycorp/gateway/common/DateUtils.kt

```diff
@@ -0,0 +1,8 @@
+package tech.relaycorp.gateway.common
+
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.util.Date
+
+fun Date.toZonedDateTime(): ZonedDateTime = ZonedDateTime.ofInstant(toInstant(), ZoneId.of("UTC"))
```

Comment:
Shouldn't this be `ZoneId.systemDefault()` instead of `ZoneId.of("UTC")`? I've been bitten by this before and I only realised when my tests broke in the transition from GMT to BST :smile:  But maybe Android is different. Would be good to check.

I suspect we may have to do something like:

```
ZonedDateTime.ofInstant(toInstant(), ZonedDateTime.systemDefault()).withZoneSameInstant(ZoneId.of("UTC"))
```

... in which case we may want to rename the method to `toZonedDateTimeUTC`.

Reply:
All dates are coming from outside the app so far. And I'm assuming all come in UTC. I store them under a `Date` class, which does not have a timezone, just stores milliseconds. So here I keep assuming there's no conversion going on, it's all still UTC. What we need to be careful is doing a plain `Date()`, cause it will be using the local timezone, and not UTC.

Reply:
Just refactored everything to stop using `Date`s to make our lives easier.

Reply:
Nice, even better!